src/wasm/asm-wasm-builder.cc

```diff
@@ -288,14 +294,29 @@ class AsmWasmBuilderImpl : public AstVisitor {
                                            static_cast<uint32_t>(index.size()));
       } else {
         if (is_set_op_) {
-          current_function_builder_->AppendCode(kExprSetLocal, false);
+          if (var->IsContextSlot()) {
+            current_function_builder_->AppendCode(kExprStoreGlobal, false);
+          } else {
+            current_function_builder_->AppendCode(kExprSetLocal, false);
+          }
           is_set_op_ = false;
         } else {
-          current_function_builder_->AppendCode(kExprGetLocal, false);
+          if (var->IsContextSlot()) {
+            current_function_builder_->AppendCode(kExprLoadGlobal, false);
+          } else {
+            current_function_builder_->AppendCode(kExprGetLocal, false);
+          }
         }
         LocalType var_type = TypeOf(expr);
         DCHECK(var_type != kAstStmt);
-        AccessTemp(LookupOrInsertLocal(var, var_type));
+        if (var->IsContextSlot()) {
+          uint32_t index = LookupOrInsertGlobal(var, var_type);
+          std::vector<uint8_t> index_vec = UnsignedLEB128From(index);
```

Comment:
This is kind of awkward and relatively inefficient. How about an AddLEB128() method and hiding the encoding inside there?